src/containers/ManageAccountPage/DeleteAccountForm/DeleteAccountForm.js

```diff
@@ -0,0 +1,213 @@
+          </span>
+        );
+
+        const resendPasswordLink = (
+          <span
+            className={css.helperLink}
+            onClick={handleResetPassword(currentUser.attributes.email)}
+            role="button"
+          >
+            <FormattedMessage id="DeleteAccountForm.resendPasswordLinkText" />
+          </span>
+        );
+
+        const resetPasswordLink = /* this.state.showResetPasswordMessage || */ resetPasswordInProgress ? (
+          <>
+            <FormattedMessage
+              id="ContactDetailsForm.resetPasswordLinkSent"
+              values={{
+                email: <span className={css.emailStyle}>{currentUser.attributes.email}</span>,
+              }}
+            />{' '}
+            {resendPasswordLink}
+          </>
+        ) : (
+          sendPasswordLink
+        );
+
+        const confirmClasses = classNames(css.confirmChangesSection, {
+          [css.confirmChangesSectionVisible]: deleteAccountConfirmed,
+        });
+
+        const classes = classNames(rootClassName || css.root, className);
+        const submitDisabled = invalid || inProgress || !deleteAccountConfirmed;
+
+        return (
+          <Form
+            className={classes}
+            onSubmit={e => {
+              handleSubmit(e);
+            }}
+          >
+            <FieldCheckbox
+              id="confirmDeleteAccount"
+              name="confirmDeleteAccount"
+              label={intl.formatMessage({ id: 'DeleteAccountForm.checkboxLabel' })}
+              value="deleteAccountSelected"
+              useSuccessColor
+            />
+            <div className={confirmClasses}>
+              <H4 as="h3" className={css.confirmChangesTitle}>
+                <FormattedMessage id="DeleteAccountForm.confirmDeleteTitle" />
+              </H4>
+              <p className={css.confirmChangesInfo}>
+                <FormattedMessage
+                  id="DeleteAccountForm.confirmDeleteInfo"
+                  values={{ marketplaceName }}
+                />
+                <br />
+                {
+                  <FormattedMessage
+                    id="DeleteAccountForm.resetPasswordLink"
+                    values={{ resetPasswordLink }}
+                  />
+                }
+              </p>
+
+              <FieldTextInput
+                className={css.password}
+                type="password"
+                name="currentPassword"
+                id={formId ? `${formId}.currentPassword` : 'currentPassword'}
+                autoComplete="current-password"
+                label={passwordLabel}
+                placeholder={passwordPlaceholder}
+                validate={passwordValidators}
+                customErrorText={passwordErrorText}
+              />
+            </div>
+            <div className={css.bottomWrapper}>
+              {// TODO: refine
+              passwordErrorText ? null : <ErrorMessage error={deleteAccountError} />}
```

Comment:
Perhaps, the ErrorMessage component should render null if the error is something that's handled elsewhere.